File: MATTHEW/leetcode/AIcoding/3.DeepSeek-v3/cpp_solutions/3393-count-paths-with-the-given-xor-value.cpp
class Solution {
public:
    int countPaths(TreeNode* root, int target) {
        unordered_map<int, int> prefix;
        prefix[0] = 1;
        int result = 0;
        dfs(root, 0, target, prefix, result);
        return result;
    }

    void dfs(TreeNode* node, int current_xor, int target, unordered_map<int, int>& prefix, int& result) {
        if (!node) return;

        current_xor ^= node->val;
        result += prefix[current_xor ^ target];
        prefix[current_xor]++;

        dfs(node->left, current_xor, target, prefix, result);
        dfs(node->right, current_xor, target, prefix, result);

        prefix[current_xor]--;
    }
};

File: leetcode/AIcoding/5.Qwen-3/go_solutions/3393-count-paths-with-the-given-xor-value.go
package main

func countPaths(root *TreeNode, k int) int {
    var result int
    path := make([]int, 0)

    var dfs func(node *TreeNode)
    dfs = func(node *TreeNode) {
        if node == nil {
            return
        }
        path = append(path, node.Val)
        currentXOR := 0
        for i := len(path) - 1; i >= 0; i-- {
            currentXOR ^= path[i]
            if currentXOR == k {
                result++
            }
        }
        dfs(node.Left)
        dfs(node.Right)
        path = path[:len(path)-1]
    }

    dfs(root)
    return result
}